register.php

```diff
@@ -24,9 +24,7 @@
         $userError = 'Sorry this is not a valid username';
     } else { $userError = '';}
 
-    if (isset($_POST["password"])) {
-        $passwordError = passwordHint($password);
-    }
+    $passwordError = passwordHint($password);
 
     if (empty($userError) && empty($passwordError) && empty($emailError)) {
```

Comment:
$emailError will always be emty as you have nothing that can set it anymore. your check needs to lok at if you have a valid Email object inside $newEmail